node/lib/Logger.ts

```diff
@@ -20,62 +22,89 @@ export class Logger implements ILogger {
 
-        if (Logger.flag === true && filename ) {
-            let fileTransport = new winston.transports.File({
+        if (Logger.flag === true && filename) {
+            this.fileTransport = new winston.transports.File({
                 filename: filename,
                 handleExceptions: true,
                 level: this.globalLogLevel,
             });
 
             this.logger.configure({
                 transports: [
-                    consoleTransport,
-                    fileTransport,
+                    this.consoleTransport,
+                    this.fileTransport,
                 ],
             });
 
-            this.transportList.push(fileTransport);
+            this.transportList.push(this.fileTransport);
             Logger.flag = false;
         }
     }
 
-   public error(msg: string, logObject?: any): void {
-        this.logger.error(msg, logObject);
+   public error(msg: string, logObject?: any, loggingMetadata?: string): void {
+        this.logger.error(msg, logObject, loggingMetadata);
     }
 
-    public warn(msg: string, logObject?: any): void {
-        this.logger.warn(msg, logObject);
+    public warn(msg: string, logObject?: any, loggingMetadata?: string): void {
+       if(typeof(logObject === 'undefined') && typeof(loggingMetadata) !== 'undefined')
+       {
+           this.logger.error(msg, loggingMetadata);
+       }
+       else if (typeof(logObject !== 'undefined') && typeof(loggingMetadata) !== 'undefined')
+       {
+           this.logger.error(msg, logObject);
+       }
+       else if (typeof(logObject === 'undefined') && typeof(loggingMetadata) === 'undefined')
+       {
+           this.logger.error(msg);
+       }        
+       else
+       { 
+           this.logger.warn(msg, logObject, loggingMetadata);
+       }
     }
 
-    public info(msg: string, logObject?: any): void {
-        this.logger.info(msg, logObject);
+    public info(msg: string, logObject?: any, loggingMetadata?: string): void {
+        this.logger.info(msg, logObject, loggingMetadata);
     }
 
-    public verbose(msg: string, logObject?: any): void {
-        this.logger.verbose(msg, logObject);
+    public verbose(msg: string, logObject?: any, loggingMetadata?: string): void {
+        this.logger.verbose(msg, logObject, loggingMetadata);
     }
 
-    public debug(msg: string, logObject?: any): void {
-        this.logger.debug(msg, logObject);
+    public debug(msg: string, logObject?: any, loggingMetadata?: string): void {
+        this.logger.debug(msg, logObject, loggingMetadata);
     }
 
     public getConfiguredTransports(): Array<any> {
         return this.transportList;
     }
 
+    public addTransport(transportObject: any): void {
+        this.logger.add(transportObject);
+
+        this.transportList.push(transportObject);
+    }
+
+    public removeTransport(transportObject: any): void {
+        this.logger.remove(transportObject);
+
+        this.transportList.pop();
```

Comment:
fixed

---
In reply to: [105339784](https://github.com/openT2T/opent2t/pull/38#discussion_r105339784) [](ancestors = 105339784)